test/tests/functional/pbs_node_rampdown_cset.py

```diff
@@ -0,0 +1,156 @@
+            if mom.is_cpuset_mom():
+                no_csetmom = False
+        if no_csetmom:
+            self.skipTest("Skip on cluster without cgroup cpuset system.")
+
+        # Various node names
+        self.n0 = self.moms.values()[0].shortname
+        self.n1 = self.moms.values()[1].shortname
+        self.n2 = '%s[0]' % (self.n1,)
+        self.n3 = '%s[1]' % (self.n1,)
+
+        # skip if second mom has less than two NUMA vnodes
+        try:
+            self.server.expect(NODE, {'state': 'free'},
+                               id=self.n3, max_attempts=10)
+        except PtlExpectError:
+            self.skipTest("Second mom has less than 2 vnodes")
+
+        a = {'state': 'free', 'resources_available.ncpus': (GE, 1)}
+        self.server.expect(VNODE, {'state=free': 4}, op=EQ, count=True,
+                           max_attempts=10, interval=2)
+
+        self.pbs_release_nodes_cmd = os.path.join(
+            self.server.pbs_conf['PBS_EXEC'], 'bin', 'pbs_release_nodes')
+
+        # number of resource ncpus to request initially
+        ncpus = self.server.status(NODE, 'resources_available.ncpus',
+                                   id=self.n3)[0]
+        ncpus = int(ncpus['resources_available.ncpus'])
+        self.ncpus2 = ncpus / 2 - 1
+
+        # expected values upon successful job submission
+        self.job1_schedselect = \
+            "1:ncpus=1:mem=2gb+1:ncpus=%d:mem=2gb+" % self.ncpus2 + \
+            "1:ncpus=%d:mem=2gb" % self.ncpus2
+        self.job1_exec_host = "%s/0+%s/0*%d+%s/1*%d" % (
+            self.n0, self.n1, self.ncpus2, self.n1, self.ncpus2)
+        self.job1_exec_vnode = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)+" % (self.n2, self.ncpus2) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n3, self.ncpus2)
+
+        # expected values after release of vnode
+        self.job1_schedsel1 = \
+            "1:ncpus=1:mem=2097152kb+1:ncpus=%d:mem=2097152kb" % (self.ncpus2)
+        self.job1_exec_host1 = \
+            "%s/0+%s/0*%d" % (self.n0, self.n1, self.ncpus2)
+        self.job1_exec_vnode1 = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n2, self.ncpus2)
+
+    def test_release_nodes_on_cpuset_sis(self):
+        """
+        Submit job that will use cpus on two NUMA vnodes on second mom,
+        goes in R state. Use pbs_release_node to release one of the NUMA
+        vnodes. The NUMA vnode and its resources get released. Submit a
+        second job requesting the released NUMA vnode. Job should run and
+        a new cpuset gets created that uses the released cpus.
+        """
+        # Submit a job that uses second mom's two NUMA nodes, in R state
+        a = {'Resource_List.select':
+             'ncpus=1:mem=2gb+ncpus=%d:mem=2gb+ncpus=%d:mem=2gb' %
+             (self.ncpus2, self.ncpus2), 'Resource_List.place': 'vscatter'}
```

Comment:
Can't this potentially end up being scheduled on the non-NUMA nodes too?  The test doesn't specifically request that vnode...maybe it should?

Reply:
It should request a specific NUMA vnode and has been changed accordingly.